lib/Promise.php

```diff
@@ -5,6 +5,7 @@
 /**
  * Representation of the future value of an asynchronous operation.
  *
+ * @template TValue
```

Comment:
Why do we need both?

Reply:
Yep, phpstan seems happy with just `template-covariant` (i don't know what the difference is)